### ISSUE TITLE
Bug in HTML Entity characters

### ISSUE THREAD

**Issue by mjablecnik** [schultek/jaspr#33]
I want to add into my page some `&nbsp;` or other [HTML entity characters](https://www.w3schools.com/html/html_entities.asp).

I have this code:
```
import 'package:jaspr/components.dart';
import 'package:jaspr/jaspr.dart';
import 'package:jaspr/styles.dart';

class BasicApp extends StatelessComponent {
  @override
  Iterable<Component> build(BuildContext context) sync* {
    yield Header(text: 'Hello World', size: HeaderSize.h2);

    yield RichText(children: [
      TextSpan(
        text: "Don't have an account?",
        styles: Styles.text(
          color: Color.named('black'),
          fontWeight: FontWeight.w500,
          fontSize: 14.pt,
        ),
      ),
      Text("&nbsp;", rawHtml: false),
      TextSpan(
        text: "Sign up",
        styles: Styles.text(
          color: Color.named('blue'),
          fontWeight: FontWeight.w500,
          fontSize: 14.pt,
        ),
      )
    ]);
  }
}
```

And after build I see this: 
![image](https://user-images.githubusercontent.com/2686906/184000264-382aca3e-4ac0-4a15-9252-41432928a8fe.png)

When I set rawHtml to true:
![image](https://user-images.githubusercontent.com/2686906/184001400-8970b2dc-aab6-4d4c-9e31-f8d2c687e608.png)
so I don't see my previous component there:
![image](https://user-images.githubusercontent.com/2686906/184001798-e81b1263-5481-4daa-ba65-e106cc995ee1.png)

Is some bug in builder?

Code of new jaspr components which I am using in example above is here: https://github.com/mjablecnik/jaspr/tree/jaspr-components/packages/jaspr/lib/src/ui/components

**Comment by schultek** [comment]
Not really a bug but a currently unsupported feature: you cannot have a rawHatml text together with other sibling elements, since the builder does not know where the raw html ends and the sibling nodes begin. A rawHtml will therefore override any other sibling nodes.

You can fix this by wrapping the rawhtml in e.g. a span.

**Comment by mjablecnik** [comment]
Ok thank you. So I added `rawHtml` also for my TextSpan component and now when I try:
![image](https://user-images.githubusercontent.com/2686906/184083174-7e93aff8-91e5-44ef-ac9a-d0961eebf6cf.png)
I see:
![image](https://user-images.githubusercontent.com/2686906/184083216-e7de7645-8800-4735-bf63-48eaf554b900.png)
When try:
![image](https://user-images.githubusercontent.com/2686906/184083309-c98367f4-6748-4125-8b6d-d289bba8bf56.png)
I see:
![image](https://user-images.githubusercontent.com/2686906/184083352-5d0a1e41-160c-4a81-8dd5-ad40d4aaa4ce.png)
When try:
![image](https://user-images.githubusercontent.com/2686906/184083450-42e4129f-3ed7-4e7a-bfac-498e85256430.png)
I see:
![image](https://user-images.githubusercontent.com/2686906/184083492-2bea8123-c498-4fd9-a794-fd36598014f5.png)
When try:
![image](https://user-images.githubusercontent.com/2686906/184083600-1b2a9fc6-a49f-4606-975c-a73a4d542156.png)
I see:
![image](https://user-images.githubusercontent.com/2686906/184083642-7eaa19ff-2dc4-481e-bdc3-daa7ddcd538b.png)
When try:
![image](https://user-images.githubusercontent.com/2686906/184084274-6f1c759d-02c5-4be9-a072-9a4bc9b0ead7.png)
I see:
![image](https://user-images.githubusercontent.com/2686906/184084317-23ea829e-c918-45be-9fc6-98548cba41e5.png)

And now when try:
![image](https://user-images.githubusercontent.com/2686906/184083919-751e76a3-a1ad-4d1e-ab8a-f49576201e4c.png)
I see:
![image](https://user-images.githubusercontent.com/2686906/184083965-1d0d63b3-3326-4203-85b7-5a204ce6214d.png)

So sometime wrapping into `span` works and sometime doesn't...

**Comment by schultek** [comment]
how is `TextSpan` implemented?

**Comment by mjablecnik** [comment]
@schultek You can see it here: https://github.com/mjablecnik/jaspr/blob/jaspr-components/packages/jaspr/lib/src/ui/components/texts.dart

**Comment by mjablecnik** [comment]
This video probably help you to see where is the bug: 

https://user-images.githubusercontent.com/2686906/184507755-b4559b11-ab05-4187-b895-3b0b09ff5852.mp4

**Comment by schultek** [comment]
I'm still investigating this, its really strange.

**Comment by schultek** [comment]
But thanks, this helps. 👍

**Comment by schultek** [comment]
I pushed a fix to `develop` & `wip/v0.2`
